src/Action.hs

```diff
@@ -257,7 +258,8 @@ class (MonadError ActionExcept m) => ActionPersist m where
         either (throwError . ActionPersistExcept) pure $ runExcept (runReaderT q db)
 
 class ActionRandomPassword m where
-    mkRandomPassword :: m InitialPassword
+    mkRandomPassword     :: m InitialPassword
+    mkForgottenPassToken :: m PassForgottenToken
```

Comment:
We should replace this with a `MonadRandom` instance in `Action.Implementation`.  Then we wouldn't have to touch the monad internals for these changes at all.

If you want to do this before the password changes in a separate PR: great.  If not, could you leave a FIXME here?  (Or is there one already?)